deps/wazuh_testing/wazuh_testing/vulnerability_detector.py

```diff
@@ -25,6 +26,45 @@ def callback_detect_vulnerability_scan_sleeping(line):
     return match.group(1) if match is not None else ""
 
 
+def callback_detect_update_from_year_result(line):
+    """
+    Check if the selected provider was added with the correct year to update sinze
+    Also, check if modulesd have failed due to a invalid year selected as update_from_year value
+
+    returns a triple:
+    ("result", "provider", "update_from_value")
+    result can be: "invalid_year", "provider_added"
+    """
+    invalid_year_msg = r'.*wazuh-modulesd.*wmodules-vuln-detector.*ERROR: Invalid content for \'update_from_year\' option at module \'vulnerability-detector\''
+    invalid_year_match = re.match(invalid_year_msg, line)
+
+    if invalid_year_match is not None:
+        return "invalid_year", None, None
+
+    provider_added_msg = r'.*wazuh-modulesd.*Added (.*) feed.*Update since: (.*)\.'
+    provider_added_match = re.match(provider_added_msg, line)
+
+    if provider_added_match is not None:
+        return "provider_added", provider_added_match.group(1), provider_added_match.group(2)
+
+    return False
+
+
+def callback_download_redhat_feed_from_year(line):
+    download_redhat_from_msg = r'.*wazuh-modulesd.*vulnerability-detector.*download.*access\.redhat\.com.*after=(.*)-.*-.*'
+    download_redhat_from_match = re.match(download_redhat_from_msg, line)
+
+    download_nvd_from_msg = nvd_check_feed_download_regex
+    download_nvd_from_match = re.match(download_nvd_from_msg, line)
+
+    if download_redhat_from_match is not None:
+        return download_redhat_from_match.group(1)
+    elif download_nvd_from_match is not None:
+        return download_nvd_from_match.group(1)
+    else:
+        return False
```

Comment:
```suggestion
        return None
```

This will keep consistency in the return values.

Reply:
That wouldn't work, the way this callback is used is: match each line of the log until return something different from False. I mean, you need to return false if nothing has matched so the log file is completely read.